### ISSUE TITLE
"gyp" is a racial slur

### ISSUE THREAD

**Issue by ghost** [nodejs/node-gyp#2168]
The word "gyp" means "to cheat or swindle". It's a shortened version of "gypsy" a slur for Roma people who are commonly maligned as cheaters and swindlers.

Some references:

- https://en.wiktionary.org/wiki/gyp
- https://www.urbandictionary.com/define.php?term=gyp
- https://www.npr.org/sections/codeswitch/2013/12/30/242429836/why-being-gypped-hurts-the-roma-more-than-it-hurts-you

**Comment by v4dkou** [comment]
https://en.wikipedia.org/wiki/GYP_(software)

**Comment by HDVinnie** [comment]
> The word "gyp" means "to cheat or swindle". It's a shortened version of "gypsy" a slur for Roma people who are commonly maligned as cheaters and swindlers.
> 
> Some references:
> 
> * https://en.wiktionary.org/wiki/gyp
> * https://www.urbandictionary.com/define.php?term=gyp
> * https://www.npr.org/sections/codeswitch/2013/12/30/242429836/why-being-gypped-hurts-the-roma-more-than-it-hurts-you

Get a life

**Comment by erbridge** [comment]
Just because it's an acronym, doesn't make what the acronym spells ok. Until today, I had no idea it even was an acronym. This isn't the first time this sort of thing has come up. See the current discussions around `master` in Git and the recent ones about GIMP that resulted in the Glimpse fork.

**Comment by v4dkou** [comment]
@erbridge There's undoubtedly a trend, this recent one about `allopenissues` also comes to mind.
https://jira.atlassian.com/browse/JRASERVER-65811

For the record, in that case the decision was ruled against the change. Despite the fact that there's ample references to the possible interpretations of the word and them not being gender-neutral, there was no consensus that while somebody might take offence in that, the technical term is in fact offensive by itself.

If you feel that there's a need for change, I think it'd be beneficial for the community as a whole to provide more context to show how this `node-gyp` issue differs from `allopenissues`.

**Comment by mmarchini** [comment]
Sharing some context: `gyp` was created by Google many years ago to be the Chromium build system. Node.js adopted it as building V8 (which is part of both Node.js and Chromium) with it was a lot easier than writing and porting every change to `make` or other build system. Google deprecated it and replaced it with GN in 2016, but Node.js didn't move to GN as the effort is too big and general consensus is to move to CMake in the future (which is also a huge effort). Recently a [fork of `gyp`](https://github.com/nodejs/gyp-next) was created to apply any patches Node.js might need until the efforts to move to CMake are finished. `node-gyp` is a wrapper around that build system, which is why it is named that way (although arguably the name is not so meaningful and can be confusing for folks who are not familiar with `gyp` as a build system).

Changing the name without a strategy is not feasible as it would likely cause a disruption in the ecosystem, breaking many packages. Also, changing the name of this project alone wouldn't be of much use, as [gyp-next](https://github.com/nodejs/gyp-next) would still have the name, and so would other projects like [node-pre-gyp](https://www.npmjs.com/package/node-pre-gyp), not to mention one of the extensions used by the build system is `.gyp`. So any changes in name here would be a much bigger effort, and someone would need to volunteer to lead it.

One plus side (technically speaking) of changing the name IMO is that we could choose a package name that is build system agnostic, and it could be used as a front-end package to make migration between build systems easier.

Reopening as I believe this is worth at least discussing. But just as recent discussions around the `master` name as default git branch, this will need a technical strategy to migrate without ecosystem disruption before any decision is made. Also, maybe it makes sense to move this to [nodejs/gyp-next](https://github.com/nodejs/gyp-next), since, as I mentioned before, any changes would not be limited to this project alone.

cc @nodejs/community-committee as it might be of your interest

**Comment by v4dkou** [comment]
For completeness sake, here are some other potentially offensive build system names.

### Rake
https://github.com/ruby/rake
> Rake (top definition)
> A long-handled gardening tool that can also mean an immoral pleasure seeker.

https://www.urbandictionary.com/define.php?term=Rake

### Buck
https://github.com/facebook/buck
> Buck (2nd definition)
> 3.n: slang for a young black man 
> 4.n: a male deer 

https://www.urbandictionary.com/define.php?term=buck

These projects should probably have same problems regarding renaming the tool (disruption in the ecosystem, breaking many packages, existing CI/CD environments, etc.)., so when evaluating cost vs. benefit, I would discuss it with these communities as well.

Feel free to open an issue in corresponding repositories for Rake and Buck.

**Comment by NarutoSage** [comment]
Honestly they shouldn't have to change the name of the project, its been that way for quite some time just because it by the imagination  resembles something somewhat of a slur if you connect distant dots, doesn't mean it should be changed and if you had to connect those dots to make a concern about something thats never been both the philosophy of the name of the project is completely stupid, and you should spend more time trying to make rockets at that point.

**Comment by jesseduffield** [comment]
This is obviously going to be an emotionally charged thread so from the get-go, I hope a healthy discussion takes place. 

Reading through [this grammarphobia post](https://www.grammarphobia.com/blog/2019/03/gypsy.html) it seems that the status of Gypsy as a slur against the Romani is disputed, even within Romani culture (emphasis mine):
> Some people who identify themselves as ethnically Roma (also called Romani or Romany) are offended by “Gypsy,” and most standard dictionaries have reservations about using it *to mean Roma*. On the other hand, some Roma people don’t mind being called “Gypsies” and others even embrace the term.

This [story](https://www.sbs.com.au/topics/life/feature/romani-road-australias-gypsy-culture) from 2016 uses the term Gypsy liberally, as does one of the subjects of the piece who says:
> I've always been incredibly proud of my Gypsy heritage

Consider this [story](https://eachother.org.uk/i-am-a-gypsy-i-want-my-grand-children-to-be-proud/) from this year titled ‘I’m A Gypsy – I Want My Grandchildren To Be Proud Of That Fact’. The subject of that piece who works for Gypsy and [Traveller Empowerment (GATE) Essex](https://www.facebook.com/GaTEssex/) explicitly says:

> I am a Gypsy challenging inequalities and prejudice to my people in any way that I can.

> If that’s politically campaigning with a board in Parliament Square, online or through mentoring – however I can help my community.

> I say ‘Gypsy’ because there’s a lot of beliefs that that word isn’t liked by us anymore – it is.

The fact that the Gypsy and Traveller Empowerment organisation, an advocacy group for the Romani, uses that word in their title, as does the [Gypsy Lore Society](http://www.gypsyloresociety.org/) tells me that the time has not yet come to cancel the term.

If node-gyp mentioned the Romani in the readme alongside a stereotypical Gypsy as a mascot, thus implying equivalence, we would definitely have a problem. If no mention of the Romani was made, we would have less of a problem but eyebrows might understandably be raised at the mascot. What the repo in fact has is no mascot, no mention of the Romani, and the only reason that 'gyp' appears in the name is because it's an acronym for Generate Your Projects. I think that the case for a rename will be far stronger after Gypsy and Traveller Empowerment and the Gypsy Lore Society change their own names.

But I'm not an expert so I'm happy to be swayed otherwise :)

**Comment by Mooninaut** [comment]
@jesseduffield "Gypsy" is arguably not a slur. "Gyp" is widely considered a slur, so your (well cited) discussion of whether the term "Gypsy" is offensive in an of itself is not particularly relevant to this issue. See https://www.telegraph.co.uk/news/uknews/law-and-order/5483053/Judge-accused-of-using-racial-slur-against-gypsies.html for example.

**Comment by artsyca** [comment]
This issue is bound to arise in many contexts. It looks like OP's account has been deleted, I'm not sure what that portends but I'd be interested to know if OP was personally offended or if this is based on hypothetical concerns of potential offence.

**Comment by dball** [comment]
Just going to observe that every time I see this name show up in a build failure, it makes me think less of everyone involved. Ignorance is one thing; knowing harm is another.

**Comment by ussjoin** [comment]
I opened a PR to fix this issue (https://github.com/nodejs/node-gyp/pull/2211) and @rvagg closed it as a duplicate, which makes no sense. Could someone please explain how a PR with passing tests dupes an issue?

**Comment by mmarchini** [comment]
@ussjoin thank you for taking an interest in this issue. I suggest reading my comment above (https://github.com/nodejs/node-gyp/issues/2168#issuecomment-661202736) to understand the technical challenges and unintended consequences of changing this package name (tl;dr it would be a huge breaking change for the entire Node.js ecosystem). As @rvagg also pointed out in the PR, the `gyp/` directory is vendored from https://github.com/nodejs/gyp-next and any changes in that directory should happen on the upstream repository instead, but note that changes there would have the same unintended consequences.

With that being said, we have an ongoing initiative to deprecate gyp on Node.js and move to a different build toolchain, and while this effort will probably take a couple years it also means that the ecosystem cost of renaming `node-gyp`/`gyp-next` won't be worth it (as it will take time to get it right anyway, and by the time we get it right we'll be changing to another build toolchain).

**Comment by ussjoin** [comment]
@mmarchini Just so I'm sure I understand you: your position is that it "won't be worth it" to remove a racial slur from the core of NodeJS for an unbounded period of at least several years, despite the notes upthread in this issue? That seems to be at odds with the Node community's [Code of Conduct](https://github.com/nodejs/admin/blob/master/CODE_OF_CONDUCT.md).

**Comment by jesseduffield** [comment]
@ussjoin I think the origin of the name matters quite a bit: if the name is in reference to Gypsies or theft, that would make the case for removing it stronger. As it stands: it's a three-letter acronym for something completely unrelated. I don't think the onus is on the repo maintainers to think about all the ways somebody might take offence to an acronym, especially when a company as progressive as Google came up with the name in the first place and nobody has expressed concern about it until now.

**Comment by ussjoin** [comment]
@jesseduffield I'm going to just leave "a company as progressive as Google" aside for others, and simply respond to "nobody has expressed concern until now:"

Scrolling up in this very thread, I see several other people "expressing concern." I'm sure they're not the only ones; this only came to my attention because I was installing a tool written in NodeJS and this tool crashed horrifically, which caused me to wonder "I'm sorry, what on earth is this thing?" While many people capable of using NodeJS tools may not be capable of submitting PRs, *I* am, so I stepped up. I can't speak for everyone else who ought to have done in the past, but this issue is here, now, and as much as you clearly don't want to "think about all the ways somebody might take offense to an acronym," that doesn't make it right; it just makes it convenient for you to ignore the issue.

**Comment by MylesBorins** [comment]
@ussjoin FWIW I don't think I would characterize what @mmarchini said as "won't be worth it" but rather "this will break a TON of stuff and we need a clear plan for the ecosystem".

I think that a plan that simply removed gyp from node altogether and replaces the build system is one that is VERY appealing to me. That being said we can't make breaking changes on LTS meaning that at the very soonest we would have to wait for v14.x to be end of life before this could truly be removed and gone.

**Comment by ussjoin** [comment]
@MylesBorins To be clear, I was quoting @mmarchini:

> the ecosystem cost of renaming node-gyp/gyp-next won't be worth it

It seems like a series of commenters asked you to create a "clear path for the ecosystem," and you declined. I've now provided a PR to fix this repository and allow other repositories to fix themselves. Seems like a path!

**Comment by MylesBorins** [comment]
I agree with the assessment of 

> the ecosystem cost of renaming node-gyp/gyp-next won't be worth it

In particular I was trying to point out that the appropriate move forward is to move to a place where we can simply drop gyp altogether and replace it with a different build system.

> It seems like a series of commenters asked you to create a "clear path for the ecosystem," and you declined

One thing I think is worth keeping in mind is that this project, as well as node, is primarily run by volunteers. We have limited bandwidth and limited time. There are many issues, both technical and cultural, that we care a great deal about but don't have the time to do. For example I would love to see [fetch in core](https://github.com/nodejs/node/issues/19393), for change [master to main](https://github.com/nodejs/node/issues/33864),  and genuinely for us to stop using gyp (both as a slur and as a technology).

That being said simply flipping the switch is going to cause significant ecosystem fallout, and this is not theoretical. It is not as simple as `s/gyp/nnabt` as much as I personally would like it to be. Gyp is deeply integrated into the build system of node.js core and of the entire native module ecosystem. We have to carefully plan how to deprecate and move off it (or rename it) and that will end up rolling out over many years as Node.js 14 doesn't EOL until 2023.

I respect your energy and passion for this, I truly do, but I hope that you can approach this assuming that many of us here are attempting to operate in good faith and want to make these changes... but that there is a significant amount of labor required to do it.

**Comment by jesseduffield** [comment]
> and as much as you clearly don't want to "think about all the ways somebody might take offense to an acronym," that doesn't make it right; it just makes it convenient for you to ignore the issue.

I'm surprised somebody supposedly compassionate about other people's feelings would say something so callous.

You yourself thought 'what on earth is this thing', and then you found that it was an acronym. Were you still offended after learning that? If I get offended by something and then discover that I misunderstood the intention and context, I'm typically no longer offended. Even in those times that I'm still offended, I don't pin it on the person saying the thing, I just mark it up as a misunderstanding. You're ascribing malevolence where there is none.

**Comment by mmarchini** [comment]
I'll ask everyone to refrain from personal attacks in this issue, as those are unproductive to the discussion.

**Comment by ussjoin** [comment]
> You yourself thought 'what on earth is this thing', and then you found that it was an acronym. Were you still offended after learning that? If I get offended by something and then discover that I misunderstood the intention and context, I'm typically no longer offended. Even in those times that I'm still offended, I don't pin it on the person saying the thing, I just mark it up as a misunderstanding. You're ascribing malevolence where there is none.

I ascribed no malevolence to the original package creators; simply ignorance. That's why I made a PR---to provide a comparatively easy fix to the issue. I wasn't even aware of the existence of this thread until my PR was closed as a dupe of it. Malevolence comes in defending the package name when many people before me (even in this thread) have pointed out the issue.

**Comment by jesseduffield** [comment]
Again I take issue with your charge of malevolence, as I would say it's quite clear that everybody in this thread is acting in good faith, but I think belabouring the point would distract from discussing the long-term solution here.

**Comment by MylesBorins** [comment]
I'd like to request that we stop debating about whether the phrase is offensive or if the etymology matters. The meaning of words change over time, and words can hold charge and power unintended by those who wielded them. I don't think we do anything but create an unempathetic environment when we attempt to explain away the emotions of others with logic + reasoning.

node-gyp is unfortunately named and WILL BE offensive to some individuals. That is not and should not be a subject of debate and attempting to do so is honestly a waste of our time and only serves to create division.

@ussjoin I very much appreciate you submitting a pull-request to help try and move this forward. I did a very quick scan and noticed one very obvious example of how the regex replace method it appears you took here will not be sufficient as a broad solution to this problem. https://github.com/nodejs/node-gyp/pull/2211/files#diff-26b681bd4dac10f25d5293c7595c435bR37

There is a function called loadConfigGypi which remains unamed while you have renamed all the references to `.gypi` files to `.nnabti`. A quick scan of GitHub shows over [450k files referring to the .gypi extension](https://github.com/search?q=.gypi&type=Code). While we can come up with solutions that deprecate or replace .gypi long term, we can't simply stop supporting it altogether. Many moons ago [we dropped support for an api in node.js that was unfortunately named](https://github.com/nodejs/node/pull/3747). (Trigger warning self-harm). We did so in stages which included leaving the original API as an alias with a deprecation warning. In fact deprecations in Node.js have a [policy for deprecations](https://github.com/nodejs/node/blob/master/doc/api/deprecations.md) that we will likely have to follow if we want to remove / deprecate even just the `.gypi` extension.

For example our central build script in [common.gypi](https://github.com/nodejs/node/blob/master/doc/api/deprecations.md) and you will fine [90+ refrencess to .gypi](https://github.com/nodejs/node/search?q=.gypi&unscoped_q=.gypi) throughout our source code. This is the type of complicated detangled we have to deal with.

Again, I appreciate your enthusiasm, but there is not relatively easy fix to this and implying otherwise is IMHO not operating in good faith.

**Comment by MylesBorins** [comment]
@jesseduffield I'd also like to request that you please refrain from continuing to engage if you are going to debate the validity of the complaint. If you would like to work towards solutions that would be awesome, but the general direction and tone of this conversation has gone slightly off the rails and the way you are personally engaging is not helping to de-escalate this.

**Comment by jesseduffield** [comment]
@MylesBorins apologies, I'll leave you to it :)

**Comment by devsnek** [comment]
Strong agree that our time would better be spent migrating away from node-gyp than working to rename it.

**Comment by benjamingr** [comment]
@ussjoin 👋 

First of all thank you for engaging, championing this change and starting to do the technical work in that PR. I think there are some misconceptions, I am going to write down things as I understand them and please let me know (you or other parties) if I got anything wrong:

 - The term gyp (which node did not pick) is unfortunate since it might be considered a racial slur. This is something that is not contested and we agree with. Everyone in the project so far is OK with changing the name `gyp` in principle and some people (like Myles above) feel strongly that we in fact should do it.
 - In addition there is a considerable amount of technical debt around gyp and Node wants to replace it with a different meta-build system altogether. This is a large concern and some people think it might not be worth it (which mostly means they are not willing to put in the large amount of effort for this to be done in a way that will not break a lot of user code).
 - Replacing gyp is a big undertaking that involves a lot of breakage. In particular [this comment by Myles](https://github.com/nodejs/node-gyp/issues/2168#issuecomment-684256232) gives a good summary of why.
 - People in Node in fact do care enough to undertake these sort of projects (namely Myles, Tierney and a few others were engaged in the renaming of the main branch from 'master'). Small renames happen all the time. 

In particular, [opening a PR](https://github.com/nodejs/node-gyp/pull/2211) without first talking to the project or people involved (they're friendly and usually communicative, I promise) isn't a great way to interact with the project because it invites a lot of miscommunication (like we saw here). I can see why it would look like the changes are relatively easy but please bear with us and assume good faith. 

I encourage you to instead reach out to the people in this thread and ask how to help and how you can promote this. I am sure that we can align both our goals here (as little ecosystem breakage as possible and renaming the unfortunate terminology). Please keep in mind that comments like [this one](https://github.com/nodejs/node-gyp/issues/2168#issuecomment-684180908) that isn't assuming Mary is acting in good faith when she comments aren't helpful in progressing this issue. The Node.js project is committed to not be complicit in racism and while we live in a complicated world we _are_ trying and we would appreciate help.

**Comment by mmarchini** [comment]
@benjamingr great summary, I'd just like to point out that:

> In addition there is a considerable amount of technical debt around gyp and Node wants to replace it with a different meta-build system altogether. This is a large concern and some people think it might not be worth it (which mostly means they are not willing to put in the large amount of effort for this to be done in a way that will not break a lot of user code).

Is not accurate anymore (it might've been a couple years ago). There's general consensus that we should move to a different meta build system and I don't remember anyone objecting to it in a long time. We even spun this as a strategic initiative recently.

**Comment by artsyca** [comment]
While we're being semantically hypersensitive we ought to rename JavaScript itself because the acronym `JS` has inappropriate connotations relating to underage girls

**Comment by benjamingr** [comment]
@artsyca Hey.

If you are bothered by this and are open to convincing other people of the importance of renaming JS in our code base or doing the actual work of making sure the acronym isn't present and are willing to collaborate and gain consensus - I am 100% willing to support this change. I am thankfully not sure what the inappropriate connotations are.

As a side note - please don't mix issues about naming you don't like in the project. We are willing to have a conversation about unfortunate or problematic naming, learn and correct ourselves. The naming issue here (gyp) is unrelated to the issue you are discussing and I would prefer it if you opened a separate issue if you feel strongly enough. 

I cannot stress enough - we take complains regarding naming seriously and we take the project and the culture seriously. People in the project are bound by a code of conduct and a lot of us want to do better with regards to problematic naming.

I would also like you to ask you to self-moderate "semantically hypersensitive", you don't get to dismiss other people here outright even if you are a project member or leadership. Please read [our code of conduct](https://github.com/nodejs/admin/blob/master/CODE_OF_CONDUCT.md) if you are not sure what is appropriate here and you're welcome to reach out if not sure.

**Comment by ussjoin** [comment]
@benjamingr This thread provides a pretty great reason why someone wouldn't want to reach out in private to maintainers:
* The PR was immediately closed.
* When I asked why, it was reopened, then immediately downvoted by @rvagg.
* The thread is filled with maintainers saying that it's not worth the effort, then walking that back.
* Not to mention the other people jumping in to tell me how terrible I am, like @artsyca and @jesseduffield (leaving aside the other people who mocked the original requesters in this issue, months before I independently came across the same issue).

Regarding breakage: yep, it's true that I intentionally broke loading of the old, slur-named files. If you wanted to add a bridge to pick those up as well for the interim, that'd be fine. Instead it sounds like this whole thing is just being downvoted as "too hard, we'll get to replacing it eventually (but not until at least 2023?) and you should just deal with it until then."

So: I won't be reaching out in private. If you, as you said, would appreciate help, then I think you and the rest of the Node team ought to reach out to the other people you've driven away upthread, as well as actually starting the work and declaring a "by X date this work will be done." For my part, while I tried to help by creating the PR in the first place, I think I've gotten my answer---and it makes me pretty sad that the Node community would rally around protecting a racial slur in the core of the language system.

**Comment by MylesBorins** [comment]
@ussjoin in my personal opinion you seem to be willfully misrepresenting what's going on in this thread.

* Maintainers are not saying it is not worth the effort of making this change. There is an existing effort to [replace the build system](https://github.com/nodejs/TSC/issues/901) which would effectively deprecate gyp and remove it from the tree entirely. If we can get this work done before Node.js 16 we have an identical timeline to permanent removal of gyp if we were to rename.
* The date of 2023 is when the 14.x release line goes end of life. We cannot make new breaking changes to 14.x and thus would be unable to completely remove gyp or gypi from that release line until then.
* We could potentially look at changing the name of gyp + gypi in 14 in a non-breaking way, but doing so in an LTS release would be difficult to do.
* Myself and others have pointed out behavior of other commenters in this thread as inappropriate and moderated as neccessary.

There are many people in this thread offering to work with you, but when you mischaracterize our intentions and what we are trying to do it makes it incredibly difficult to assume you are coming in here and operating in good faith.

> it makes me pretty sad that the Node community would rally around protecting a racial slur in the core of the language system.

I don't see this happening in this issue at all. What I do see are multiple individuals saying they agree with you, that we can and do want to do better, but that we can't do that at the cost of legitimately breaking many many many systems. We have active work going on to improve this, it is not the solution you are proposing, but it is a long term viable solution that would have a similar timeline to what you are proposing.

I really do want to believe that you are operating in good faith here and simply very passionate about this topic, but from where I am sitting it would appear that you are willfully ignoring what folks are saying and misrepresenting opinions in a way that calls into question the ethics of our team. That is something I am extremely uncomfortable with and do not appreciate.

**Comment by MylesBorins** [comment]
@ussjoin one more thing worth mentioning. I disagree with the immediate closure of your original PR and quick -1. I think that this is definitely worth having a conversation about. If replacing the build system in node is going to take a significant amount of time we could potentially reduce harm in the mean time through renaming... but that effort will need to be a bit more thoughtful than just a blanket rename which is how the PR is currently implemented.

If you are willing to work with me on it I would spend time with folks on our end to figure out what the requirements for a rename would be and how we could potentially do it in tangent to replacing the build tool so that we don't have a legacy tool with a name that is a slur. I can't commit to doing the labor, I simply don't have time right now, but I can help with building consensus and defining what work would need to be done.

**Comment by devsnek** [comment]
Maybe to clarify a bit: We aren't saying we don't want to get rid of node-gyp, we're saying we believe the most effective way to get rid of node-gyp is the major-cycle deprecation of it. I don't think anyone in this thread wants node-gyp to continue to be used, it's an awful build system with an awful name.

**Comment by mmarchini** [comment]
Adding to @MylesBorins last comment: the Node.js project has hundreds of collaborators from all around the world, and opinions/points of view are not homogeneous across the project. This heterogeneity can sometimes lead to mishaps (like what happened with your PR being closed 5 minutes after being opened), and those mishaps don't represent the perspective of all collaborators, just the person who made the action/commented.

I don't think the PR should land _as is_ either (that's not a -1 from me, but a "Request for Changes"). At the very least changes to `gyp/` need to be moved to the gyp-next repository, and _in a first PR_ it needs to keep working on existing projects (meaning the `gyp` binary needs to stay, and so does the file extensions) without the maintainers needing to make any changes. Preferably we would need to somehow run the changes on the PR against a list of popular native modules on npm, like we do for many changes on Node.js via CITGM, but I don't think we have the infrastructure to target specific packages atm. If you or someone else is willing to do that work, I'm happy to coordinate with the new build system initiative.

**Comment by artsyca** [comment]
@ussjoin this is a very complex issue that we simply don't have the civility and decorum to navigate not to mention knowing the wider implications of following down this sort of moralistic pathway but I digress.

I would kindly request my name be removed from your comment above as I never mentioned you once and instead I'm trying to point out that this issue and this approach to this issue has far wider implications than it seems anyone seems to be aware of.

**Comment by ussjoin** [comment]
@artsyca Since your comment was as follows (screenshot from my email, since it was removed from this thread, but I think it's important for the historical record), I decline. If "corporate, salty, immature, and pretentious" is your response to "please stop using a racial slur as the name of a core Node system" then I have serious concerns about any organization in which you might participate.

![IMG_8772](https://user-images.githubusercontent.com/32540/91906682-778c0480-ec5d-11ea-852c-852fd055acf6.jpg)

**Comment by ussjoin** [comment]
> @ussjoin in my personal opinion you seem to be willfully misrepresenting what's going on in this thread.

Fine; then prove me wrong. I'll look forward to the announcement of the change. I don't see continuing value in contributing further code.

**Comment by artsyca** [comment]
Exactly my point, sir. The term JS is an innuendo to be sure and an image search is enough to convince anyone that the meaning is not limited only to javascript but the attitude that some commenters are bringing is rather defensive and dogmatic which doesn't help anyone.

I'm trying to point out that this is a huge rabbit hole and we're not prepared for where it leads and I'm speaking as someone who only stumbled into this issue because this blasted package wouldn't compile properly, so imagine my disdain for pursing it further.

**Comment by v4dkou** [comment]
@ussjoin I've initially noticed this thread when I went to GitHub to research some issues with building a dependency using node-gyp.
This build system has enough problems as is, I especially _like_ fiddling around with nvm to align the ~stars~ versions right, so that, i.e. react-native with Realm on old NodeJS versions and a backend with OpenCV on newer NodeJS work fine.

We basically have two options here:

- Rename node-gyp, making the ecosystem even more fragile and fiddly, drawing a lot of negative attention towards the cause, because all the frustration from DX issues around the world will be re-directed to cursing political correctness and your PR personally.
- Ride the wave and push for moving to a different meta build system.

The first option is a lose-lose scenario, the second is win-win.

Why aren't we discussing the second option here?
It'd be a shame if we waste all the energy on the clearly suboptimal option and the movement dies out for nothing.

**Comment by rachelnicole** [comment]
I think the best option would be to close this issue and to create a new issue in good faith about what the future holds instead of arguing about semantics and intent. No one here is discounting that the term is negative no matter the original intention nor are they arguing for not doing anything.

**Comment by MylesBorins** [comment]
@artsyca and @v4dkou I've already asked politely above for folks to please stop debating the validity of a name change.

The phrase is offensive, full stop. If you continue to derail conversation I will be personally asking for moderators to limit your ability to participate in this organization.

@ussjoin it is absolutely within your right to not continue to give us volunteer labor. With that said the only thing I would like to ask for is the benefit of the doubt that we are going to try and make things right.

I have a distinct feeling that the initial immediate closure of your PR got things off on the wrong foot here. With that said, expecting an ecosystem breaking change like this to land in a single PR without any changes isn't very realistic. There are multiple people here willing to work with you on the name change if you want to drive it. The same folks are putting in effort to solve the problem in a different way, removing gyp.

I'm genuinely sorry that you have had such a bad experience engaging here, but I really hope you can see they it isn't very reasonable to bring us a half-baked solution and expect a group of volunteers to finish it for you.

We have a solution we'll be working on, we want the same outcome. If you want to collaborate with us on the solution you've suggested we are here to work with you

**Comment by mmarchini** [comment]
FYI for the efforts of removing gyp altogether we have an issue on TSC for discussion (https://github.com/nodejs/TSC/issues/901) and we recently created a repository to have more organized discussions and planning (https://github.com/nodejs/build-toolchain-next). I don't believe this repository is the appropriate place to discuss removing gyp from core and as the default native addon toolchain since both discussions go way beyond this repository.

**Comment by v4dkou** [comment]
@MylesBorins I understand that you are in a tough position and you have to bring order so that we come to a productive conclusion.
I want the same outcome (being less hate in the world), and I am trying to bring another perspective on the topic.

> @artsyca and @v4dkou I've already asked politely asked for folks to please stop debating the validity of a name change.
> 
> The phrase is offensive, full stop. 

I am not debating the offensiveness here, I am just chiming in about the potential backlash, which could bring even more hate towards the cause.

Let me rephrase:

- When things break, people get upset
- To figure things out, they browse GitHub issues
- When they find this PR, they connect the name change to them being upset

In other words, disruption produces more hate, the change won't win over people who'll curse our names for taking part in it.

Does it sound plausible or am I the only one imagining the salty developer venting his frustration with a combination of personal attacks on contributors to the change and racial slurs? Haven't we all seen this co-worker who is loud and angry about things not working as intended?
And I wouldn't dismiss this person as regressive either, as emotional responses can be triggered by stress alone.

Please let me know if my experiences align with yours in any way or am I tilting at windmills here.

> If you continue to derail conversation I will be personally asking for moderators to limit your ability to participate in this organization.

Sorry, but that feels uncalled for.
If that's what you genuinely believe is right, I will just leave myself, as I honestly don't get why my contribution is not welcome.

**Comment by artsyca** [comment]
And I, as a poor beleaguered software purist understand the concept of `DRY` extends to our thought and conversation patterns and can imagine this same debate repeating itself incessantly in various contexts up to and including the naming of the JavaScript language itself and I simply can't afford to see the entire industry go up in arms in every single instance, however well intentioned.

Going by [wikipedia:GYP (Disambiguation)](https://en.wikipedia.org/wiki/Gyp), the word is indeed a perjorative but it also the IATA code for Gympie airport in Queensland and a common name for a genus of flowers in addition to a proper name referring to actual people and a song by Bobby Darrin.

I sincerely don't feel it would be sweeping this under the rug to simply state clearly that the acronym is just three letters that bear an unfortunate connotation to an antiquated perjorative of unknown origin from the 19th century, and I'm all for renaming the node package to something more generic.

Furthermore, the word can also mean a servant or domestic helper, and in this context it seems very apt.

Anyway, I don't want to be banned by the moderators and this is what I'm taking issue with, given than we all know better, why do our conversations always devolve into naming, shaming, blaming and inflaming -- when did software become so unfun?

I'm out, you'll hear no further comments from me, I'm grateful to have made my little mark on the history of software. Goodbye.

**Comment by benjamingr** [comment]
@artsyca 

> Anyway, I don't want to be banned by the moderators and this is what I'm taking issue with, given than we all know better, why do our conversations always devolve into naming, shaming, blaming and inflaming -- when did software become so unfun?

As a moderation team member - bringing up the validity of the request and etymology after being requested to stop multiple times and then cancelling other peoples' opinions and experience is exactly the sort of thing that could get you banned.

We (people like me, Myles, Mary and other project contributors) take this (and @ussjoin's comments) seriously. It's fine if you don't - but please: if you are not a regular contributor to Node feel free to not engage in these sort of discussions if you goal is to get Node not to explore renaming things.

Here's the thing: for a large amount of people the naming of GYP isn't a big deal, apparently you are one of those people. That's fine. I am one of those people too and am pretty ignorant on this topic. If however other people feel strongly about an issue in Node and want to discuss it so we can resolve it - why not discuss it with them and resolve it? There might be a way that avoids ecosystem breakage (there often is).

Let's not assume this is a zero sum game.

**Comment by artsyca** [comment]
Ben, this is a meta-meta discussion by now but here goes, may you have mercy upon me and give me the benefit of doubt.

Human beings are an interesting bunch, we have mastered (are we allowed to use this word in this context?) the periodic table, the electron, to a large extent the natural world, and yet we still haven't figured out how to assemble in groups to have a civilized conversation, or if we did we've lost key subtleties of the art.

A lot of us when we communicate, we don't precisely say what we mean but rather we try to paint pictures with words, others think out loud and wait to hear or see their own words before understanding what they're trying to say. Some of us speak with mathematical precision bordering on the pedantic while others use subtle references and cues.

What doesn't change however is that none of us are actually speaking to anyone but our own selves and everything we write needs to be interpreted by the receiving party based on context, assumptions, prejudice, and several layers of emotional, intellectual and instinctual conditioning.

Somewhere along the way, we invented these machines that are meant to increase our capability of communicating and as always the medium remains the message, at its core the message is the audience. We are all like messages trapped inside of bottles trying our best to break free and deliver the payload to the necessary party, all while struggling against incredible odds and at heavy cost to ourselves -- no wonder a lot simply choose to remain silent and 'bottle it up'

I would say the discussion at hand is a very important one to me, a very big deal indeed, for whatever reason, even though I'm not a contributor and a mere guest in this environment. You'd think we'd have invented a way where everyone could speak their piece asynchronously without fear of reprisals or reprimand, especially when they happen to stroll into a flame war which subsequently follows them home to their inbox and they truly want to understand the forces at play and help make sure every voice is heard. There's a reason we say 'people over processes' after all, and this process seems to place a heavy emphasis on the linearity of the thread over the ability of any old buffoon to put their brainfart down for the record.

To me, this discussion is the perfect example of all our failings and redeeming qualities as human beings (for you can't separate one from the other) and I'm hoping against hope that someone else will see it this way and also notice the irony of trying to take a hammer to an issue that requires a touch much lighter than a feather. As well it seems like an opportunity to have this kind of meta-meta discussion but alas the chilling effects are prevalent and the forecast looks like more chilling effects on the way.

I understand your reference to game theory, and I totally understand that there can be a win-win situation here. Can you believe me when I tell you that I really didn't mean to be a 'very bad boy' and despite several wrist slaps I actually have some valuable contributions to make if I wasn't politely and not-so-politely being shown the door, and there are probably thousands of people just like me wondering what to do with their human capital and how they can come to the aid of the Roma people and every other objectified, marginalized and stigmatized member of this so-called society we're building for ourselves?

I've gone and deleted all my past comments as I realize now how triggering they may have seemed. If you get anything out of this it's that we ought to treat others like time travellers from another era who are out of sync with our own reality and give them some way to vent their frustrations without taking it so gosh darned personally.

**Comment by acoolstraw** [comment]
keep it, it'll just save more people from using this horrifying piece of software

**Comment by ninjarogue** [comment]
This is ridiculous.. don't you have anything better to do than squabble over such a trivial thing as the acronym gyp which in no way has anything to do with the actual slur? Why do you expect everyone to adhere to your feeble conscience?
